doe/tasks/azure/cli.yml

```diff
@@ -0,0 +1,44 @@
+- name: Install Microsoft GPG Key for rpm
+  ansible.builtin.rpm_key:
+    state: present
+    key: https://packages.microsoft.com/keys/microsoft.asc
+  become: true
+  become_user: root
+  when: ansible_distribution == "CentOS" or ansible_distribution == "RedHat"
+
+- name: Install Microsoft Repo from rpm package
+  ansible.builtin.yum:
+    name: https://packages.microsoft.com/config/rhel/9.0/packages-microsoft-prod.rpm
+    state: present
+  become: true
+  become_user: root
+  when: ansible_distribution == "CentOS" or ansible_distribution == "RedHat"
+
+- name: Install Microsoft GPG Key for apt
+  ansible.builtin.apt_key:
+    state: present
+    url: https://packages.microsoft.com/keys/microsoft.asc
+  become: true
+  become_user: root
+  when: ansible_distribution == "Ubuntu"
```

Comment:
`when: ansible_distribution == "Ubuntu" or ansible_distribution == "Debian"` might be better.